omnipaxos_ui/src/app.rs

```diff
@@ -50,31 +54,47 @@ pub(crate) struct App {
     pub(crate) followers_progress: Vec<f64>,
     /// The accepted_idx of all the followers. Idx is the pid of the node.
     pub(crate) followers_accepted_idx: Vec<u64>,
-    pub(crate) color_generator: ColorGenerator,
+    // Color generator for the current node, the first color is for the current node.
+    // pub(crate) color_generator: ColorGenerator,
 }
 
 impl App {
     pub(crate) fn with(config: UIAppConfig) -> Self {
         let max_peer_pid = config.peers.iter().max().unwrap();
         let max_pid = *std::cmp::max(max_peer_pid, &config.pid) as usize;
+        let mut color_generator = ColorGenerator::new(COLORS.to_vec());
+        let current_node = Node {
+            pid: config.pid,
+            configuration_id: config.configuration_id,
+            ballot_number: 0,
+            connectivity: 0,
+            color: *color_generator.next_color(),
+        };
+        let mut nodes: Vec<Node> = config
+            .peers
+            .clone()
+            .into_iter()
+            .map(|pid| Node {
+                pid,
+                color: *color_generator.next_color(),
+                ..Default::default()
+            })
+            .collect();
+        nodes.push(current_node.clone());
+        nodes.sort_by(|a, b| a.pid.cmp(&b.pid));
```

Comment:
We should first push(), then sort, and then map the colors so that everything is consistent even if we run dashboard on different nodes.